amplify/src/main/java/com/github/stkent/amplify/tracking/predicates/PredicateMap.java

```diff
@@ -0,0 +1,100 @@
+/**
+ * Copyright 2015 Stuart Kent
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.github.stkent.amplify.tracking.predicates;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.github.stkent.amplify.tracking.GenericSettings;
+import com.github.stkent.amplify.tracking.interfaces.IEvent;
+import com.github.stkent.amplify.tracking.interfaces.IEventCheck;
+import com.github.stkent.amplify.tracking.interfaces.ILogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * Created by bobbake4 on 11/13/15.
+ */
+public abstract class PredicateMap<T> extends ConcurrentHashMap<IEvent, List<IEventCheck<T>>> {
+
+    private final ILogger logger;
+    private final GenericSettings<T> settings;
+    private final Context applicationContext;
+
+    public abstract void eventTriggered(@NonNull final IEvent event);
+
+    public PredicateMap(ILogger logger, GenericSettings<T> settings, Context applicationContext) {
+        super();
+        this.logger = logger;
+        this.settings = settings;
+        this.applicationContext = applicationContext;
+    }
+
+    public ILogger getLogger() {
+        return logger;
+    }
+
+    public GenericSettings<T> getSettings() {
+        return settings;
+    }
+
+    public Context getApplicationContext() {
+        return applicationContext;
```

Comment:
So the getSettings is the only one not used but the getters were a PMD requirement. I should change them all to protected and I guess I can remove the getSettings until we need it in a sub class.